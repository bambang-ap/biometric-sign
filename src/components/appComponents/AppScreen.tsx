import React from 'react';
import {SafeAreaView, View} from 'react-native';

import {AppScreenProps} from '@appTypes/propsType.type';
import {classNames} from '@utils';

export default function AppScreen({children, className}: AppScreenProps) {
  return (
    <SafeAreaView className={classNames('flex-1 bg-white', className)}>
      <View className="flex-1 px-2 pt-2">{children}</View>
    </SafeAreaView>
  );
}
