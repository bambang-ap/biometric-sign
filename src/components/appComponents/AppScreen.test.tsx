import React from 'react';

import {render} from '@testing-library/react-native';
import FullMockComponent from '__mocks__/@components';
import renderer from 'react-test-renderer';

import AppScreen from './AppScreen';
describe('AppScreen Component', () => {
  const TestComponent = () => {
    return (
      <FullMockComponent>
        <AppScreen />
      </FullMockComponent>
    );
  };

  test('AppScreen render correctly', () => {
    render(<TestComponent />);
  });

  test('AppScreen snapshot', () => {
    const wrapper = renderer.create(<TestComponent />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
