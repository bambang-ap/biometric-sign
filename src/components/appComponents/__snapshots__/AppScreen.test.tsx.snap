// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppScreen Component AppScreen snapshot 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "paddingLeft": 8,
          "paddingRight": 8,
        },
        {
          "paddingTop": 8,
        },
      ]
    }
  />
</RCTSafeAreaView>
`;
